--- r1000_v1.05/src/r1000_n.c~	2006-11-15 04:50:32.000000000 +0200
+++ r1000_v1.05/src/r1000_n.c	2007-12-12 12:34:34.137131255 +0200
@@ -1718,7 +1718,11 @@
 
 static int __init r1000_init_module (void)
 {
-	return pci_module_init (&r1000_pci_driver);	// pci_register_driver (drv)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,10)
+	return pci_register_driver(&r1000_pci_driver);
+#else
+	return pci_module_init(&r1000_pci_driver);
+#endif
 }
 
 static void __exit r1000_cleanup_module (void)
